File: VB/MVVMExpenses/Models/Category.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.ComponentModel.DataAnnotations

Namespace MVVMExpenses.DataModels
    Public Class Category
        <Key, Display(AutoGenerateField:=False)>
        Public Property ID() As Long
        <Required, StringLength(30, MinimumLength:=5), Display(Name:="CATEGORY")>
        Public Property Name() As String
        <EnumDataType(GetType(TransactionType)), Display(Name:="TRANSACTION TYPE")>
        Public Property Type() As TransactionType
        Public Overrides Function ToString() As String
            Return Name & " (" & Type.ToString() & ")"
        End Function
    End Class
End Namespace

File: CS/MVVMExpenses/Models/Category.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;

namespace MVVMExpenses.DataModels {
    public class Category {
        [Key, Display(AutoGenerateField = false)]
        public long ID { get; set; }
        [Required, StringLength(30, MinimumLength = 5)]
        [Display(Name = "CATEGORY")]
        public string Name { get; set; }
        [EnumDataType(typeof(TransactionType))]
        [Display(Name = "TRANSACTION TYPE")]
        public TransactionType Type { get; set; }
        public override string ToString() {
            return Name + " (" + Type.ToString() + ")";
        }
    }
}
